src/app/public/modules/radio/radio.component.ts

```diff
@@ -236,6 +242,12 @@ export class SkyRadioComponent implements OnDestroy, ControlValueAccessor {
     return this._selectedValue;
   }
 
+  /**
+   * Selection box component uses this to update selected states.
+   * @internal
+   */
+  public deselect: Subject<void> = new Subject<void>();
```

Comment:
Instead of creating a new internal property, what if we added a more useful `checkedChange` observable to both the radio and checkbox components?

**Selection box component:**

```
@Component({
  selector: 'sky-selection-box',
  template: `
  <div
    class="sky-selection-box"
    tabindex="0"
    [ngClass]="{ active: isChecked }"
  >
    ...
    <ng-content
      select="sky-checkbox,sky-radio"
    ></ng-content>
    ...
  </div>
  `
})
export class SkySelectionBoxComponent implements OnInit {

  @Input()
  public control: SkyCheckboxComponent | SkyRadioComponent;

  public active = false;

  public ngOnInit(): void {
    this.control.checkedChange.subscribe(checked => {
      this.active = checked;
    });
  }
}
```

**Checkbox/radio code:**
```
@Input()
public set checked(value: boolean) {
  ...
  this._checked = value;
  this._checkedChange.next(this._checked);
  ...
}

@Output()
public get checkedChange(): Observable<boolean> {
  return this._checkedChange.asObservable();
}

private _checkedChange = new BehaviorSubject<boolean>(this._checked);
```

**Consumer code:**
```
<sky-selection-box
  [control]="checkbox1"
>
  <sky-icon></sky-icon>

  <sky-selection-box-header>
    Heading here.
  </sky-selection-box-header>

  <sky-selection-box-description>
    Description here.
  </sky-selection-box-description>

  <sky-checkbox
    #checkbox1
  ></sky-checkbox>
</sky-selection-box>
```

Reply:
This has been done.